### PR TITLE
Shortcut & Mouse Icon for Toggling Mouse Note Entry

### PR DESCRIPTION
**Pull request by worldwideweary**
Resolves: Musescore user's desire for shortcut to enable/disable mouse entry more quickly in relation to #323 
Figured I'd also throw-in a toolbar icon up in there:
image

In order to get it working on my machine, I had to manually add it like so:

https://github.com/Jojo-Schmitz/MuseScore/assets/7139517/9696c492-37bf-479d-b4aa-52c8c8239891

While this was going on the other day, I figured I'd also get some auto-highlighting effects to show what *will be* selected prior to selection. It is an advanced preference option by default as "transparent" and then if set, it will apply in front of everything. If the alpha is 255 or fully opaque, it will highlight "behind" everything, not coloring any objects but only highlighting behind in z-order. I don't know if anyone would find that useful, but it's kind of nice as an option. That's another issue/pr though sometime possibly

### CODE REVIEW

File: mscore/events.cpp [hunk 1]
```diff
@@ -572,7 +572,19 @@ void ScoreView::mousePressEvent(QMouseEvent* ev)
...
                   if (ev->button() == Qt::RightButton)
                         _score->inputState().setRest(!restMode);
-                  _score->putNote(editData.startMove, ev->modifiers() & Qt::ShiftModifier, ev->modifiers() & Qt::ControlModifier);
+                  if (MScore::disableMouseEntry) {
+                        if (auto el = elementAt(editData.pos)) {
+                              if (!el->isStaffLines()) {
+                                    _score->select(el);
+                                    if (el->isNote() || el->isRest())
+                                          _score->inputState().moveInputPos(el);
+                                    else changeState(ViewState::NORMAL);
```
Comment (Jojo-Schmitz) [inline mscore/events.cpp:581]: Formatting...

File: global/settings/types/preferencekeys.h [hunk 2]
```diff
@@ -119,6 +119,7 @@
 #define PREF_SCORE_NOTE_PLAYONCLICK                         "score/note/playOnClick"
 #define PREF_SCORE_NOTE_DEFAULTPLAYDURATION                 "score/note/defaultPlayDuration"
 #define PREF_SCORE_NOTE_WARNPITCHRANGE                      "score/note/warnPitchRange"
+#define PREF_SCORE_NOTE_INPUT_DISABLE_MOUSE_INPUT           "ui/score/noteEntry/disableMouseEntry"
```
Comment (Jojo-Schmitz) [inline global/settings/types/preferencekeys.h:122]: That's already in, why again? Forgot to rebase?

File: libmscore/mscore.cpp [hunk 3]
```diff
@@ -95,6 +95,7 @@ QColor  MScore::frameMarginColor;
 QColor  MScore::bgColor;
 QColor  MScore::dropColor;
 bool    MScore::warnPitchRange;
+bool    MScore::disableMouseEntry;
```
Comment (Jojo-Schmitz) [inline libmscore/mscore.cpp:98]: same here

File: libmscore/mscore.h [hunk 4]
```diff
@@ -341,6 +341,7 @@ class MScore {
       static QColor frameMarginColor;
       static QColor bgColor;
       static bool warnPitchRange;
+      static bool disableMouseEntry;
```
Comment (Jojo-Schmitz) [inline libmscore/mscore.h:344]: and here

File: libmscore/shadownote.cpp [hunk 5]
```diff
@@ -115,7 +115,7 @@ bool ShadowNote::computeUp() const
 
 void ShadowNote::draw(QPainter* painter) const
       {
-      if (!visible() || !isValid())
+      if (!visible() || !isValid() || MScore::disableMouseEntry)
```
Comment (Jojo-Schmitz) [inline libmscore/shadownote.cpp:118]: and here

File: mscore/events.cpp (hunk 1 above)
Comment (Jojo-Schmitz) [reply mscore/events.cpp:581]: here too, in addition to the formatting request, which I apparently missed last time

File: mscore/musescore.cpp [hunk 6]
```diff
@@ -432,6 +433,7 @@ void updateExternalValuesFromPreferences() {
       MScore::playRepeats = preferences.getBool(PREF_APP_PLAYBACK_PLAYREPEATS);
       MScore::playbackSpeedIncrement = preferences.getInt(PREF_APP_PLAYBACK_SPEEDINCREMENT);
       MScore::warnPitchRange = preferences.getBool(PREF_SCORE_NOTE_WARNPITCHRANGE);
+      MScore::disableMouseEntry = preferences.getBool(PREF_SCORE_NOTE_INPUT_DISABLE_MOUSE_INPUT);
```
Comment (Jojo-Schmitz) [inline mscore/musescore.cpp:436]: here too

File: mscore/preferences.cpp [hunk 7]
```diff
@@ -220,6 +220,7 @@ void Preferences::init(bool storeInMemoryOnly)
             {PREF_SCORE_NOTE_PLAYONCLICK,                          new BoolPreference(true, false)},
             {PREF_SCORE_NOTE_DEFAULTPLAYDURATION,                  new IntPreference(300 /* ms */, false)},
             {PREF_SCORE_NOTE_WARNPITCHRANGE,                       new BoolPreference(true, false)},
+            {PREF_SCORE_NOTE_INPUT_DISABLE_MOUSE_INPUT,            new BoolPreference(false, true)},
```
Comment (Jojo-Schmitz) [inline mscore/preferences.cpp:223]: and here,
But not sure we'd still need it hidden in Advanced Preferences?

File: mscore/scoreview.cpp [hunk 8]
```diff
@@ -1021,7 +1021,6 @@ void ScoreView::setShadowNote(const QPointF& p)
       qreal mag     = score()->staff(pos.staffIdx)->mag(Fraction(0,1));
       qreal relX    = pos.pos.x() - pos.segment->measure()->canvasPos().x();
       pos.pos.rx() -= qMin(relX - score()->styleP(Sid::barNoteDistance) * mag, 0.0);
-
```
Comment (Jojo-Schmitz) [inline mscore/scoreview.cpp:1024]: and here

File: mscore/scoreview.cpp [hunk 9]
```diff
@@ -3206,7 +3205,8 @@ void ScoreView::startNoteEntry()
 
       getAction("pad-rest")->setChecked(false);
       setMouseTracking(true);
-      shadowNote->setVisible(true);
+      if (!MScore::disableMouseEntry)
+            shadowNote->setVisible(true);
```
Comment (Jojo-Schmitz) [inline mscore/scoreview.cpp:3209]: and here

File: mscore/preferences.cpp (hunk 7 above)
Comment (worldwideweary) [reply mscore/preferences.cpp:223]: Yeah, that could be removed... P.S. apparently forgot to pull from your 3.x before making a branch based on the last PR, so the second commit showed up. Hopefully isn't a problem though, but if it is I can redo it this and then make the preference set to false to not show up...

Comment (Jojo-Schmitz) [reply mscore/preferences.cpp:223]: Too late, I've just merged it ;-)

**Comment by worldwideweary** [comment]
P.S. Forgot to store the preference so that it is saved for the user who actually uses this toggle. . . will do another PR for that on the side sometime

**Pull request merged**